Team9Project/src/com/indragie/cmput301as1/ExpenseClaimTabFragment.java

```diff
@@ -0,0 +1,482 @@
+		startActivityForResult(addIntent, ADD_EXPENSE_CLAIM_REQUEST);
+	}
+	
+	/**
+	 * Starts the {@link ExpenseClaimSortActivity}
+	 */
+	private void startSortExpenseClaimActivity() {
+		Intent intent = new Intent(activity, ExpenseClaimSortActivity.class);
+		startActivityForResult(intent, SORT_EXPENSE_CLAIM_REQUEST);
+	}
+		
+	/**
+	 * Starts the {@link ManageTagsActivity}
+	 */
+	private void startManageTagsActivity() {
+		Intent manageTagsIntent = new Intent(activity, ManageTagsActivity.class);
+		startActivityForResult(manageTagsIntent, MANAGE_TAGS_REQUEST);
+	}
+	
+	/**
+	 * Starts the {@link FilterTagsActivity}
+	 */
+	private void startFilterTagsActivity() {
+		Intent filterTagsIntent = new Intent(activity, FilterTagsActivity.class);
+		filterTagsIntent.putExtra(FilterTagsActivity.TAG_TO_FILTER, filteredTagsList);
+		startActivityForResult(filterTagsIntent, FILTER_TAGS_REQUEST);
+	}
+
+
+	/**
+	 * Sets the list used in ListModel to the returned list of expense claims from the intent. 
+	 * @param data The intent to get the list of expense claims from. 
+	 */
+	/**
+	 * Sets the list used in ListModel to the returned list of expense claims from the intent. 
+	 * If tag is in the filteredTagsList, we have to update the list to accommodate for the changes.
+	 * Displays the filteredListModel instead if there are filtered tags.
+	 * @param data The intent to get the list of expense claims from. 
+	 */
+	@SuppressWarnings("unchecked")
+	private void onManageTagsResult(Intent data) {
+		ArrayList<ManageTagsActivity.TagMutation> mutations =
+				(ArrayList<ManageTagsActivity.TagMutation>)data.getSerializableExtra(ManageTagsActivity.EXTRA_TAG_MUTATIONS);
+		SparseArray<ExpenseClaim> modifiedClaims = new SparseArray<ExpenseClaim>();
+		
+		for (ManageTagsActivity.TagMutation mutation : mutations) {
+			Tag tag = mutation.getOldTag();
+			int i = 0;
+			for (ExpenseClaim claim : listModel.getItems()) {
+				if (!claim.hasTag(tag)) continue;
+				
+				switch (mutation.getMutationType()) {
+				case DELETE:
+					claim.removeTag(tag);
+					if (filteredTagsList.contains(tag) && !filteredTagsList.isEmpty()) {
+						filteredTagsList.remove(tag);
+					}
+					break;
+				case EDIT:
+					int tagIndex = claim.getTags().indexOf(tag);
+					claim.setTag(tagIndex, mutation.getNewTag());
+					if (filteredTagsList.contains(tag) && !filteredTagsList.isEmpty()) {
+						int selectedIndex = filteredTagsList.indexOf(tag);
+						filteredTagsList.set(selectedIndex, mutation.getNewTag());
+					}
+					break;
+				}
+				
+				modifiedClaims.append(i, claim);
+				i++;
+			}
+			
+		}
+		
+		for (int i = 0; i < modifiedClaims.size(); i++) {
+			listModel.set(modifiedClaims.keyAt(i), modifiedClaims.valueAt(i));
+		}
+		if (!filteredTagsList.isEmpty()) {
+			setFilteredClaims();
+		}
+		
+	}
+
+	/**
+	 * Prompts the user for confirmation in response to deleting an expense
+	 * claim.
+	 */
+	public void startDeleteAlertDialog() {
```

Comment:
Looks like this wasn't merged correctly either, the implementation in master takes an integer parameter